lambdas/sqs-message-remover/tests/test-index.js

```diff
@@ -201,7 +225,16 @@ test('sqsMessageRemover lambda removes message from queue when workflow succeede
   t.is(numberOfMessages.numberOfMessagesAvailable, 0);
   t.is(numberOfMessages.numberOfMessagesNotVisible, 0);
 
-  await awsServices.sqs().deleteQueue({ QueueUrl: sqsQueues.queueUrl }).promise();
+  // Check that item does not exist in S3 and therefore throws an error
+  await t.throwsAsync(s3().getObject({
```

Comment:
Weirdly, this doesn't throw... If I call to delete the message directly, it throws. The logs demonstrate that the path is `updateSqsQueue -> deleteArchivedMessages` without erroring, but the object in the bucket actually isn't deleted.

Reply:
After looking into this, the reason why the object can still be retrieved is that deletion is eventually consistent so there isn't a guarantee that, immediately after deletion, the object is not retrievable.